core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -114,7 +128,13 @@ class Project(
   // debounces ReloadExistingFilesEvent
   private var rechecking: Cancellable = _
 
-  def handleRequests: Receive = withLabel("handleRequests") {
+  def idleMonitor(r: Receive)(implicit context: ActorContext): Receive = {
+    case i @ Idle =>
+      processIdleMessage(); r(i)
+    case i => seenWorkMessage; r(i)
+  }
+
+  def handleRequests: Receive = withLabel("handleRequests")(idleMonitor {
```

Comment:
the more canonical thing to do here would be `idleMonitor orElse handleRequests` when you set the listening state of the actor.